src/client.js

```diff
@@ -95,4 +130,55 @@ export default class Client {
   on(event, cb) {
     this.eventEmitter.on(event, cb);
   }
+
+  /* ---- HELPERS ---- */
+  get(path, options) {
+    const _options = options || {};
+    _options.method = 'get';
+    return this.fetch(path, _options);
+  }
+
+  post(path, body, options) {
+    let _options = { ...this.defaults.post };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'post';
+    return this.fetch(path, _options);
+  }
+
+  put(path, body, options) {
+    let _options = { ...this.defaults.put };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'put';
+    return this.fetch(path, _options);
+  }
+
+  patch(path, body, options) {
+    let _options = { ...this.defaults.patch };
+    _options.body = JSON.stringify(body);
+    _options = merge.recursive(true, _options, options);
+    _options.method = 'patch';
+    return this.fetch(path, _options);
+  }
+
+  delete(path, options) {
+    const _options = options || {};
+    _options.method = 'delete';
+    return this.fetch(path, _options);
+  }
+
+  upload(path, input, options) {
+    const _options = options || {};
+    _options.method = 'post';
+
+    const data = new FormData();
+    let i;
+    for (i = 0; i < input.files.length; i++) {
+      data.append('files[]', input.files[i]);
```

Comment:
Yeah, that's what I'm getting at.  Just uploading a single file is overly simple IMO, there tends to be some other data you'd want to add with the file.  That's why I'm saying it should be form* methods, which would receive an object defining the key-vals.  Any value could be a File object, or array of files.  This would make upload redundant and we wouldn't need it... you'd just `formPost('/path', {'files[]': File})`